src/checklist/Checklist.tsx

```diff
@@ -132,97 +148,142 @@ const Checklist = (props: ChecklistProps): JSX.Element => {
         };
     }, [reloadChecklist, api]);
 
+    const handleShowMoreClick = (): void => {
+        setShowMore(true);
+    };
+
+    const handleShowLessClick = (): void => {
+        setShowMore(false);
+    };
+
     return (
         <AsyncPage
             fetchStatus={fetchChecklistStatus}
             errorMessage={'Unable to get checklist.'}
             loadingMessage={''}
         >
             <>
-                {loopTags.length > 0 ? (
-                    <LoopTagWrapper>
-                        <Caption>Loop tags:</Caption>
-                        <List>
-                            {loopTags.map((loopTag) => (
-                                <List.Item key={loopTag.tagId}>
-                                    <Caption>{loopTag.tagNo}</Caption>
-                                </List.Item>
-                            ))}
-                        </List>
-                    </LoopTagWrapper>
-                ) : null}
-                {!isSigned && !allItemsCheckedOrNA
-                    ? null
-                    : isSigned && (
-                          <Banner>
-                              <Banner.Message>
-                                  This checklist is signed. Unsign to make
-                                  changes.
-                              </Banner.Message>
-                          </Banner>
-                      )}
-                {checklistDetails ? (
-                    <ChecklistWrapper>
-                        {!isSigned && (
-                            <CheckAllButton
+                {!multiSignOrVerifyIsOpen && (
```

Comment:
Implement the loop tags section as a separate component - LoopTags.tsx - now that its complexity has increased.